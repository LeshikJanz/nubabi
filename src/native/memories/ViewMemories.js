// @flow
import type { State, Memory as MemoryType } from '../../common/types';
import React, { PureComponent } from 'react';
import { compose } from 'ramda';
import { graphql, gql } from 'react-apollo';
import { connect } from 'react-redux';
import { displayLoadingState, showNoContentViewIf } from '../components';
import { isEmptyProp, mapEdgesToProp } from '../../common/helpers/graphqlUtils';
import MemoryList from './MemoryList';
import Memory from './Memory';

type Props = {
  memories: Array<MemoryType>,
  currentBabyId: string,
  onEditMemory: (id: string) => void,
};

export class ViewMemories extends PureComponent {
  props: Props;

  static fragments = {
    list: gql`
      fragment Memories on Baby {
        memories {
          edges {
            node {
              id
              ...MemoryItem
            }
          }
        }
      }
      ${Memory.fragments.detail}
    `,
  };

  render() {
    return (
      <MemoryList
        babyId={this.props.currentBabyId}
        memories={this.props.memories}
        onEditMemory={this.props.onEditMemory}
      />
    );
  }
}

export default compose(
  connect(({ babies }: State) => ({
    currentBabyId: babies.currentBabyId,
  })),
  graphql(
    gql`
    query ViewMemories($babyId: ID) {
      viewer {
        baby(id: $babyId) {
          id
          ...Memories
        }
      }
    }
    ${ViewMemories.fragments.list}

  `,
    {
      options: ({ currentBabyId }) => ({
        variables: { babyId: currentBabyId },
        fetchPolicy: 'cache-and-network',
      }),
      props: mapEdgesToProp('viewer.baby.memories', 'memories'),
    },
  ),
  showNoContentViewIf(isEmptyProp('memories')),
  displayLoadingState,
)(ViewMemories);
