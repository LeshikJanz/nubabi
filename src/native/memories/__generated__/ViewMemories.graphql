query ViewMemories($babyId: ID) {
  viewer {
    baby(id: $babyId) {
      id
      ...Memories
    }
  }
}

fragment Memories on Baby {
  memories {
    edges {
      node {
        id
        ...MemoryItem
      }
    }
  }
}


fragment MemoryItem on Memory {
  id
  title
  createdAt
  # TODO: how to combine pagination with GalleryScreen
  files {
    count
    edges {
      node {
        id
        contentType
        url

        ... on Image {
          thumb {
            url
          }
          large {
            url
          }
        }

        ... on Audio {
          duration
        }

        ... on Video {
          thumb {
            url
          }
          duration
        }
      }
    }
  }
}
