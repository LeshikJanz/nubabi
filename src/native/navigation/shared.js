import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) => (
  pathOr('Nubabi', ['params', 'babyName'], state)
);

export function getBabyTitle() {
  return { title: babyNameTitle };
}

export const getTabHeaders = (navigate) => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
