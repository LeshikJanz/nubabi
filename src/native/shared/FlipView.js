import React, { Component } from 'react';
import {
  Animated,
  TouchableOpacity,
} from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

class FlipCard extends Component {
  constructor(props) {
    super(props);
    this._flipToggleCard = this._flipToggleCard.bind(this);
    this.state = {
      animatedValue: new Animated.Value(0),
      isFlipped: true,
    };
  }

  componentDidUpdate(prevProp, prevState) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard() {
    this.setState({ isFlipped: !this.state.isFlipped });
  }

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0,   // Returns to the start
      velocity: this.props.velocity,  // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction,  // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront;
    }
    return this.props.renderBack;
  }

  render() {
    const rotateX = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <TouchableOpacity onPress={this._flipToggleCard} style={styles.animatedContainer}>
        <Animated.View
          style={[styles.animatedContainer, { transform: [{ rotateX }] }]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </TouchableOpacity>);
  }
}

FlipCard.propTypes = {
  renderBack: React.PropTypes.object.isRequired,
  renderFront: React.PropTypes.object.isRequired,
  velocity: React.PropTypes.number.isRequired,
  tension: React.PropTypes.number.isRequired,
  friction: React.PropTypes.number.isRequired,
};

module.exports = FlipCard;
