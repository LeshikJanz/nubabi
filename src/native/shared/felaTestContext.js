import React from 'react';
import { NativeFelaProvider } from '../components/FelaProvider';
import { createRenderer } from 'fela-native';
import theme from '../../common/themes/defaultTheme';

export const renderer = createRenderer();
const FelaProvider = NativeFelaProvider(renderer, theme);

const felaTestContext = component =>
  <FelaProvider>
    {component}
  </FelaProvider>;

export default felaTestContext;
