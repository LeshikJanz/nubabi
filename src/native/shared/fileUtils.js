// @flow
import { getContentTypeFromFilename } from '../../common/helpers/graphqlUtils';

export const isNewFile = (file: ?{ url: string }) => {
  if (!file) {
    return false;
  }

  return !file.url.startsWith('https://firebasestorage.googleapis.com');
};

// Parse results from react-native-image-picker into File interface inout
// (do not confuse with react-native-image-crop-picker which already gives
// us the expected result
export const fileFromImagePickerResult = file => {
  return {
    size: file.fileSize,
    name: file.fileName,
    url: file.uri,
    contentType: getContentTypeFromFilename(file.fileName),
  };
};
