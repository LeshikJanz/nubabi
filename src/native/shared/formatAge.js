import moment from 'moment';
import pluralize from 'pluralize';

const pluralizeAge = (count, unit) => {
  return [count, pluralize(unit, count), 'old'].join(' ');
};

export const formatAge = dobString => {
  const now = moment();
  const dob = moment(dobString);

  const diff = now.diff(dob, 'months');
  const shouldDisplayWeeks = diff === 0;

  if (shouldDisplayWeeks) {
    const weeks = now.diff(dob, 'week');
    return pluralizeAge(weeks, 'week');
  }

  const shouldDisplayYears = diff >= 12;

  if (shouldDisplayYears) {
    const years = now.diff(dob, 'years');
    return pluralizeAge(years, 'year');
  }

  return pluralizeAge(diff, 'month');
};
