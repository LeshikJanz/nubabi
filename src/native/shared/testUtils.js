// @flow
import testRenderer from 'react-test-renderer';
import felaTestContext, { renderer as felaRenderer } from './felaTestContext';

const felaNativeToStyleObject = cache => {
  return Object.keys(cache).reduce((acc, key) => {
    return { ...acc, ...JSON.parse(key) };
  }, {});
};

export const createTestRenderer = () => (Component: any) => {
  const component = testRenderer.create(felaTestContext(Component));

  const snapshot = component.toJSON();
  snapshot.felaRules = felaNativeToStyleObject(felaRenderer.cache);

  return snapshot;
};

export const createExpectRender = () => {
  const render = createTestRenderer();
  return (Component: any) => {
    expect(render(Component)).toMatchSnapshot();
  };
};

export const expectRender = createExpectRender();
