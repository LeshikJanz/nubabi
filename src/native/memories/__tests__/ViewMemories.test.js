import 'react-native';
import React from 'react';
import { ViewMemories } from '../ViewMemories';
import { expectRender } from '../../shared/testUtils';

const memories = require('./ViewMemories.fixture.json');

test('it renders correctly', () => {
  expectRender(<ViewMemories babyId="some" memories={memories} />);
});
