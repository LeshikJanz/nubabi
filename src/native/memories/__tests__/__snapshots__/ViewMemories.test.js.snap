// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {}

<RCTScrollView
  data={
    Object {
      "__typename": "MemoryConnection",
      "edges": Array [
        Object {
          "__typename": "MemoryEdge",
          "node": Object {
            "__typename": "Memory",
            "author": Object {
              "__typename": "User",
              "avatar": Object {
                "__typename": "Avatar",
                "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/users%2FdHY49Iz4O9UByM7ZKBqmJ7GdnX83%2Favatar?alt=media&token=ec13f678-c0c2-4412-a540-e2e58ae08154",
              },
            },
            "comments": Object {
              "__typename": "CommentConnection",
              "count": 6,
              "edges": Array [
                Object {
                  "__typename": "CommentEdge",
                  "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
                  "node": Object {
                    "__typename": "Comment",
                    "author": Object {
                      "__typename": "User",
                      "firstName": "Cullen",
                      "lastName": "Kris",
                    },
                    "createdAt": "2017-07-15T01:21:12.695Z",
                    "id": "1dc5dbd2-a591-4421-8fdf-aca5a65b4c26",
                    "text": "Sint voluptatem odit quam in nam. Veniam neque atque placeat ut provident natus quaerat cum. Debitis non sint voluptate porro. Dolor perferendis repudiandae facilis itaque ipsum itaque dolores harum.",
                  },
                },
                Object {
                  "__typename": "CommentEdge",
                  "cursor": "YXJyYXljb25uZWN0aW9uOjE=",
                  "node": Object {
                    "__typename": "Comment",
                    "author": Object {
                      "__typename": "User",
                      "firstName": "Travis",
                      "lastName": "Ritchie",
                    },
                    "createdAt": "2017-07-03T13:49:13.386Z",
                    "id": "9af48c8c-a268-46ed-a221-f4883e8c5a7e",
                    "text": "Distinctio voluptatem sapiente accusamus quibusdam quisquam voluptas facere. Repudiandae facere voluptatibus placeat blanditiis consequatur sequi. Voluptas qui dignissimos. A suscipit odit facere inventore ab repellat.",
                  },
                },
              ],
            },
            "createdAt": "2017-07-19T14:53:10.000Z",
            "files": Object {
              "__typename": "FileConnection",
              "count": 5,
              "edges": Array [
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpQKoih7HypAXCivixG%2F4DFD9E45-6A1F-43CE-BC1B-35D8E20629BF.jpg?alt=media&token=507e6408-3622-47f9-9fda-7bd85680ec16",
                  },
                },
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpQKoih7HypAXCivixG%2F42CBB834-24C3-4ACD-9B94-170B3AF30BF8.jpg?alt=media&token=0d8bd3f4-786f-420d-96a5-c43864e3034a",
                  },
                },
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpQKoih7HypAXCivixG%2F0131E309-3D46-4EFD-8E0A-684B0F502D98.jpg?alt=media&token=ebae2733-5610-4ee8-9004-5a8dc4f16521",
                  },
                },
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpQKoih7HypAXCivixG%2F1F7B7673-7B22-4CDA-96EC-93432057DFD7.jpg?alt=media&token=fd50eb5c-d128-4ddc-ac41-199f163e72bf",
                  },
                },
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "video/mp4",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpQKoih7HypAXCivixG%2F0A38D6D4-8F88-4641-98DF-665AA500A198.mp4?alt=media&token=6cc562d6-359d-4555-96f4-348886beca76",
                  },
                },
              ],
            },
            "id": "TWVtb3J5Oi1LcFFLb2loN0h5cEFYQ2l2aXhH",
            "title": "Lots of assets",
          },
        },
        Object {
          "__typename": "MemoryEdge",
          "node": Object {
            "__typename": "Memory",
            "author": Object {
              "__typename": "User",
              "avatar": Object {
                "__typename": "Avatar",
                "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/users%2FdHY49Iz4O9UByM7ZKBqmJ7GdnX83%2Favatar?alt=media&token=ec13f678-c0c2-4412-a540-e2e58ae08154",
              },
            },
            "comments": Object {
              "__typename": "CommentConnection",
              "count": 6,
              "edges": Array [
                Object {
                  "__typename": "CommentEdge",
                  "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
                  "node": Object {
                    "__typename": "Comment",
                    "author": Object {
                      "__typename": "User",
                      "firstName": "Cullen",
                      "lastName": "Kris",
                    },
                    "createdAt": "2017-07-15T01:21:12.695Z",
                    "id": "1dc5dbd2-a591-4421-8fdf-aca5a65b4c26",
                    "text": "Sint voluptatem odit quam in nam. Veniam neque atque placeat ut provident natus quaerat cum. Debitis non sint voluptate porro. Dolor perferendis repudiandae facilis itaque ipsum itaque dolores harum.",
                  },
                },
                Object {
                  "__typename": "CommentEdge",
                  "cursor": "YXJyYXljb25uZWN0aW9uOjE=",
                  "node": Object {
                    "__typename": "Comment",
                    "author": Object {
                      "__typename": "User",
                      "firstName": "Travis",
                      "lastName": "Ritchie",
                    },
                    "createdAt": "2017-07-03T13:49:13.386Z",
                    "id": "9af48c8c-a268-46ed-a221-f4883e8c5a7e",
                    "text": "Distinctio voluptatem sapiente accusamus quibusdam quisquam voluptas facere. Repudiandae facere voluptatibus placeat blanditiis consequatur sequi. Voluptas qui dignissimos. A suscipit odit facere inventore ab repellat.",
                  },
                },
              ],
            },
            "createdAt": "2017-07-17T15:10:43.000Z",
            "files": Object {
              "__typename": "FileConnection",
              "count": 2,
              "edges": Array [
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpG5XvuMxoJbYIe5pJK%2F34F466A1-ED89-41E3-8411-4A9F2E284E93.jpg?alt=media&token=f14512a1-2a56-41d5-af84-91164b75c6f2",
                  },
                },
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpG5XvuMxoJbYIe5pJK%2FA653BDA0-FCBD-4A49-BF46-15D5BA20159A.jpg?alt=media&token=5d64e435-77c7-41da-acf7-84092a0b672e",
                  },
                },
              ],
            },
            "id": "TWVtb3J5Oi1LcEc1WHZ1TXhvSmJZSWU1cEpL",
            "title": "Some",
          },
        },
        Object {
          "__typename": "MemoryEdge",
          "node": Object {
            "__typename": "Memory",
            "author": Object {
              "__typename": "User",
              "avatar": Object {
                "__typename": "Avatar",
                "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/users%2FdHY49Iz4O9UByM7ZKBqmJ7GdnX83%2Favatar?alt=media&token=ec13f678-c0c2-4412-a540-e2e58ae08154",
              },
            },
            "comments": Object {
              "__typename": "CommentConnection",
              "count": 6,
              "edges": Array [
                Object {
                  "__typename": "CommentEdge",
                  "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
                  "node": Object {
                    "__typename": "Comment",
                    "author": Object {
                      "__typename": "User",
                      "firstName": "Cullen",
                      "lastName": "Kris",
                    },
                    "createdAt": "2017-07-15T01:21:12.695Z",
                    "id": "1dc5dbd2-a591-4421-8fdf-aca5a65b4c26",
                    "text": "Sint voluptatem odit quam in nam. Veniam neque atque placeat ut provident natus quaerat cum. Debitis non sint voluptate porro. Dolor perferendis repudiandae facilis itaque ipsum itaque dolores harum.",
                  },
                },
                Object {
                  "__typename": "CommentEdge",
                  "cursor": "YXJyYXljb25uZWN0aW9uOjE=",
                  "node": Object {
                    "__typename": "Comment",
                    "author": Object {
                      "__typename": "User",
                      "firstName": "Travis",
                      "lastName": "Ritchie",
                    },
                    "createdAt": "2017-07-03T13:49:13.386Z",
                    "id": "9af48c8c-a268-46ed-a221-f4883e8c5a7e",
                    "text": "Distinctio voluptatem sapiente accusamus quibusdam quisquam voluptas facere. Repudiandae facere voluptatibus placeat blanditiis consequatur sequi. Voluptas qui dignissimos. A suscipit odit facere inventore ab repellat.",
                  },
                },
              ],
            },
            "createdAt": "2017-07-17T13:44:57.000Z",
            "files": Object {
              "__typename": "FileConnection",
              "count": 1,
              "edges": Array [
                Object {
                  "__typename": "FileEdge",
                  "node": Object {
                    "__typename": "File",
                    "contentType": "image/jpeg",
                    "url": "https://firebasestorage.googleapis.com/v0/b/nubabitest1.appspot.com/o/babies%2F-KlNutUyADmNPA9ABHtc%2Fmemories%2F-KpFn-yiPwewxMbgYF3J%2F943118D1-2D2A-45C8-AF9A-E527C9F2A0F8.jpg?alt=media&token=b15acba6-8e59-4486-bb7e-d443b86facc4",
                  },
                },
              ],
            },
            "id": "TWVtb3J5Oi1LcEZuLXlpUHdld3hNYmdZRjNK",
            "title": "Monty suntanning ",
          },
        },
      ],
    }
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
